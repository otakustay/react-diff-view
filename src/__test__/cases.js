/* In case you are looking for type of hunk, see src/Hunk/index.js */
import {parseDiff} from '..';

export const basic = `diff --git a/src/__test__/index.test.jsx b/src/__test__/index.test.jsx
index 643c2f0..7883597 100644
--- a/src/__test__/index.test.jsx
+++ b/src/__test__/index.test.jsx
@@ -21,3 +21,3 @@ describe('basic test', () => {
     test('App renders correctly', () => {
-        expect(renderer.create(<App diffText={'deff'} />).toJSON()).toMatchSnapshot();
+        expect(renderer.create(<App diffText={'diff'} />).toJSON()).toMatchSnapshot();
     });`;

export const basicHunk = parseDiff(basic)[0].hunks[0];

export const multiple = `diff --git a/src/index1.js b/src/index1.js
index e69de29..d00491f 100644
--- a/src/index1.js
+++ b/src/index1.js
@@ -0,0 +1 @@
+1
diff --git a/src/index.js b/src/index2.js
index d00491f..0cfbf08 100644
--- a/src/index2.js
+++ b/src/index2.js
@@ -1 +1 @@
-1
+2
diff --git a/src/index3.js b/src/index3.js
index 0cfbf08..e69de29 100644
--- a/src/index3.js
+++ b/src/index3.js
@@ -1 +0,0 @@
-2`;

export const multipleHunk = parseDiff(basic)[0].hunks[0];
