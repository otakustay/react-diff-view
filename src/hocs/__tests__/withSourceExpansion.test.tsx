import * as renderer from 'react-test-renderer';
import {Diff, Hunk, HunkType} from '../..';
import {basic, basicHunk} from '../../__tests__/cases';
import {withSourceExpansion} from '..';

function ComponentIn() {
    return <div />;
}

describe('withSourceExpansion', () => {
    test('returns a hoc function', () => {
        expect(typeof withSourceExpansion()).toBe('function');
    });

    test('hoc function return a component', () => {
        expect(typeof withSourceExpansion()(ComponentIn)).toBe('function');
    });

    test('render component', () => {
        const ComponentOut = withSourceExpansion()(ComponentIn);
        expect(renderer.create(<ComponentOut />).toJSON()).toMatchSnapshot();
    });
});

interface PropsUnfoldCollapsed {
    previousHunk: HunkType;
    currentHunk: HunkType;
    onClick: (start: number, end: number) => void;
}

function UnfoldCollapsed({previousHunk, currentHunk, onClick}: PropsUnfoldCollapsed) {
    const start = previousHunk ? previousHunk.oldStart + previousHunk.oldLines : 1;
    const end = currentHunk.oldStart - 1;

    /* eslint-disable react/jsx-no-bind */
    return (
        <div onClick={() => onClick(start, end)}>
            Click to expand
        </div>
    );
    /* eslint-enable react/jsx-no-bind */
}

interface PropsDiffView {
    hunks: HunkType[];
    onExpandRange: () => void;
}

function DiffView({hunks, onExpandRange}: PropsDiffView) {
    const renderHunk = (children: any[], hunk: HunkType) => {
        const previousElement = children[children.length - 1];
        const decorationElement = (
            <UnfoldCollapsed
                key={'decoration-' + hunk.content}
                previousHunk={previousElement && previousElement.props.hunk}
                currentHunk={hunk}
                onClick={onExpandRange}
            />
        );
        children.push(decorationElement);

        const hunkElement = (
            <Hunk
                key={'hunk-' + hunk.content}
                hunk={hunk}
            />
        );
        children.push(hunkElement);

        return children;
    };

    return (
        <Diff hunks={hunks} diffType="modify" viewType="split">
            {hunks => hunks.reduce(renderHunk, [])}
        </Diff>
    );
}

describe('withSourceExpansion document usage', () => {
    test('renders correctly', () => {
        const EnhancedDiffView = withSourceExpansion()(DiffView);
        expect(renderer.create(
            <EnhancedDiffView oldSource={basic} hunks={[basicHunk]} />
        ).toJSON()).toMatchSnapshot();
    });
});
