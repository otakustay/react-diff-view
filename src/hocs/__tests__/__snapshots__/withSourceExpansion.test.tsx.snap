// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSourceExpansion document usage renders correctly 1`] = `
<table
  className="diff diff-split"
  onMouseDown={[Function]}
>
  <colgroup>
    <col
      className="diff-gutter-col"
    />
    <col />
    <col
      className="diff-gutter-col"
    />
    <col />
  </colgroup>
  <div
    onClick={[Function]}
  >
    Click to expand
  </div>
  <tbody
    className="diff-hunk"
  >
    <tr
      className="diff-line diff-line-normal"
    >
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        21
      </td>
      <td
        className="diff-code diff-code-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            test('App renders correctly', () =&gt; {
      </td>
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        21
      </td>
      <td
        className="diff-code diff-code-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            test('App renders correctly', () =&gt; {
      </td>
    </tr>
    <tr
      className="diff-line diff-line-compare"
    >
      <td
        className="diff-gutter diff-gutter-delete"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        22
      </td>
      <td
        className="diff-code diff-code-delete"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
                expect(renderer.create(&lt;App diffText={'deff'} /&gt;).toJSON()).toMatchSnapshot();
      </td>
      <td
        className="diff-gutter diff-gutter-insert"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        22
      </td>
      <td
        className="diff-code diff-code-insert"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
                expect(renderer.create(&lt;App diffText={'diff'} /&gt;).toJSON()).toMatchSnapshot();
      </td>
    </tr>
    <tr
      className="diff-line diff-line-normal"
    >
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        23
      </td>
      <td
        className="diff-code diff-code-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            });
      </td>
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        23
      </td>
      <td
        className="diff-code diff-code-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            });
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`withSourceExpansion render component 1`] = `<div />`;
